### PR TITLE
[GLUTEN-9015][VL] Support array_append function

### PR DESCRIPTION
**Pull request by dcoliversun**
## What changes were proposed in this pull request?

This PR aims to support array_append function for Spark 3.4 and 3.5.

(Fixes: \#9015)

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by dcoliversun** [comment]
@PHILO-HE @rui-mo Could you please have a review for this PR? Thanks.

**Review by PHILO-HE — COMMENTED** [review]
@dcoliversun, you may need to fix CH test failure. You may need to change some CH backend code to make this function fall back at that side.

**Comment by dcoliversun** [comment]
@PHILO-HE This is updated :)

File: backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala [hunk 1]
```diff
@@ -123,6 +123,58 @@ abstract class ScalarFunctionsValidateSuite extends FunctionsValidateSuite {
...
+
+        runQueryAndCompare("select arr, num, array_append(arr, num) from tbl") {
+          checkGlutenOperatorMatch[ProjectExecTransformer]
+        }
+
+        runQueryAndCompare("select array_append(array(), NULL) from tbl") {
+          checkGlutenOperatorMatch[ProjectExecTransformer]
+        }
+
+        runQueryAndCompare("select array_append(array(NULL), NULL) from tbl") {
```
Comment (jackylee-ch) [inline backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala:150]: These are not suitable cases, as Spark would directly execute them in optimizer, can we write and read from parquet?

Comment (dcoliversun) [reply backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala]: yes, updated :)

File: backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala [hunk 2]
```diff
@@ -123,6 +123,54 @@ abstract class ScalarFunctionsValidateSuite extends FunctionsValidateSuite {
...
+    withTempPath {
+      path =>
+        Seq[(Array[Int], Int)](
+          (Array(2, 1), 0),
+          (Array(1), 1),
+          (Array(), 0),
+          (Array(1, 2, null.asInstanceOf[Int]), 1),
+          (Array(null.asInstanceOf[Int]), 1),
+          (Array(null.asInstanceOf[Int]), null.asInstanceOf[Int]),
+          (Array(), null.asInstanceOf[Int])
```
Comment (jackylee-ch) [inline backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala:136]: please add `(null, 1)` to test, as vanilla spark return null for `SELECT array_append(CAST(null as Array<Int>), 2);`

**Review by jackylee-ch — APPROVED** [review]
Left a small comment. Thanks for your work.

File: backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala (hunk 2 above)
Comment (dcoliversun) [reply backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala]: ok, added

**Comment by PHILO-HE** [comment]
Not relevant CI failure. Will merge the pr soon.
